t/22_ipaddr2.t

```diff
@@ -866,14 +825,70 @@ subtest '[ipaddr2_os_cloud_init_logs]' => sub {
             return;
     });
 
-    ipaddr2_os_cloud_init_logs();
+    ipaddr2_cloudinit_logs();
 
     for my $call_idx (0 .. $#calls) {
         note($calls[$call_idx][0] . " C-->  $calls[$call_idx][1]   $calls[$call_idx][2]");
     }
     ok((scalar @calls > 0), "Some calls to ipaddr2_ssh_internal");
 };
 
+subtest '[ipaddr2_cloudinit_create]' => sub {
+    my $ipaddr2 = Test::MockModule->new('sles4sap::ipaddr2', no_auto => 1);
+    my $cloud_init_content;
+    $ipaddr2->redefine(write_sut_file => sub {
+            $cloud_init_content = $_[1];
+            return; });
+    $ipaddr2->redefine(upload_logs => sub { return '/Faggin'; });
+
+    ipaddr2_cloudinit_create();
+
+    note("cloud_init_content:\n" .
+          "--------------------------\n" .
+          $cloud_init_content .
+          "\n--------------------------");
+
+    like($cloud_init_content, qr/nginx/, "cloud-init script is also about nginx");
+    unlike($cloud_init_content, qr/registercloudguest/, "cloud-init script does not register");
+};
+
+subtest '[ipaddr2_cloudinit_create] with scc_code' => sub {
+    my $ipaddr2 = Test::MockModule->new('sles4sap::ipaddr2', no_auto => 1);
+    my $cloud_init_content;
+    $ipaddr2->redefine(write_sut_file => sub {
+            $cloud_init_content = $_[1];
+            return; });
+    $ipaddr2->redefine(upload_logs => sub { return '/Faggin'; });
+
+    ipaddr2_cloudinit_create(scc_code => 'ABCD');
+
+    note("cloud_init_content:\n" .
+          "--------------------------\n" .
+          $cloud_init_content .
+          "\n--------------------------");
+
+    like($cloud_init_content, qr/registercloudguest.*ABCD/, "cloud-init script registers");
+};
+
+subtest '[ipaddr2_cloudinit_create] nginx_root' => sub {
+    my $ipaddr2 = Test::MockModule->new('sles4sap::ipaddr2', no_auto => 1);
+    my $cloud_init_content;
+    $ipaddr2->redefine(write_sut_file => sub {
+            $cloud_init_content = $_[1];
+            return; });
+    $ipaddr2->redefine(upload_logs => sub { return '/Faggin'; });
+
+    ipaddr2_cloudinit_create(nginx_root => 'ABCD');
+
+    note("cloud_init_content:\n" .
+          "--------------------------\n" .
+          $cloud_init_content .
+          "\n--------------------------");
+
+    #like($cloud_init_content, qr/registercloudguest.*ABCD/, "cloud-init script registers");
```

Comment:
Commented out code. Is it required?

Reply:
o yes it is, or better to say the next `ok(1)` has to be converted to something more usefull